src/pycell.rs

```diff
@@ -884,6 +937,7 @@ impl From<PyBorrowMutError> for PyErr {
 pub trait PyCellLayout<T>: PyLayout<T> {
     fn get_borrow_flag(&self) -> BorrowFlag;
     fn set_borrow_flag(&self, flag: BorrowFlag);
```

Comment:
This trait would probably have to change a bit:

```rust
pub trait PyCellLayout<T: PyClass>: PyLayout<T> {
    fn borrow_checker(&self) -> &T::BorrowChecker;
    
    // ... rest unchanged
```

Reply:
I suspect that if you do that you end up with a circular trait bound somewhere.